src/LiveQuery/ParseLiveQueryServer.js

```diff
@@ -327,6 +331,71 @@ class ParseLiveQueryServer {
       if (isSubscriptionSessionTokenMatched) {
         return Parse.Promise.as(true);
       }
+
+      // Check if the user has any roles that match the ACL
+      return new Parse.Promise(function(resolve, reject){
```

Comment:
you should probably use arrow functions in order to not need to scope capture this.

That will be safer ;)